mongo/errors.go

```diff
@@ -184,6 +184,12 @@ func (e MongocryptdError) Unwrap() error {
 	return e.Wrapped
 }
 
+type labeledError interface {
+	error
+	// HasErrorLabel returns true if the error contains the specified label.
+	HasErrorLabel(string) bool
+}
```

Comment:
This seems like a reasonable change to fix this bug.

To check my understanding:
1. `driver.Error` represents a server error. It is usually hidden from the caller. We do not want to expose `x` types directly. `errors.replaceErrors` will replace the top-level `driver.Error` with a `CommandError`, which implements `ServerError`.
2. In this case, since the `driver.Error` is wrapped in a `topology.ConnectionError` and other auth errors, it won't be replaced.

Would another alternative be to have `driver.Error` implement the full `ServerError` interface? I am thinking it may let a caller use `errors.As` on a handshake error and call other `ServerError` methods.